server/src/main/java/org/opensearch/index/store/RemoteSegmentStoreDirectory.java

```diff
@@ -848,17 +927,50 @@ public void deleteStaleSegmentsAsync(int lastNMetadataFilesToKeep, ActionListene
         }
     }
 
+    public static boolean cleanupAsync(
```

Comment:
I don't like the usage of boolean here, instead would be better to just throw the exception back to the caller